lib/apiservers/engine/backends/container.go

```diff
@@ -1567,98 +1671,65 @@ func (c *Container) containerAttach(name string, ca *backend.ContainerAttachConf
-	}
-
-	// commit the handle; this will reconfigure the vm
-	_, err = client.Containers.Commit(containers.NewCommitParamsWithContext(ctx).WithHandle(handle))
-	if err != nil {
-		switch err := err.(type) {
-		case *containers.CommitNotFound:
-			return NotFoundError(name)
-		case *containers.CommitConflict:
-			return ConflictError(err.Error())
-		case *containers.CommitDefault:
-			return InternalServerError(err.Payload.Message)
-		default:
-			return InternalServerError(err.Error())
-		}
+	if err := c.containerProxy.BindInteraction(handle, name, id); err != nil {
+		return err
 	}
 
-	clStdin, clStdout, clStderr, err := ca.GetStreams()
+	stdin, stdout, stderr, err := ca.GetStreams()
 	if err != nil {
 		return InternalServerError("Unable to get stdio streams for calling client")
 	}
-	defer clStdin.Close()
+	defer stdin.Close()
 
 	if !vc.Config.Tty && ca.MuxStreams {
 		// replace the stdout/stderr with Docker's multiplex stream
-		if ca.UseStdout {
-			clStderr = stdcopy.NewStdWriter(clStderr, stdcopy.Stderr)
-		}
 		if ca.UseStderr {
-			clStdout = stdcopy.NewStdWriter(clStdout, stdcopy.Stdout)
+			stderr = stdcopy.NewStdWriter(stderr, stdcopy.Stderr)
+		}
+		if ca.UseStdout {
+			stdout = stdcopy.NewStdWriter(stdout, stdcopy.Stdout)
 		}
 	}
 
 	actor := CreateContainerEventActorWithAttributes(vc, map[string]string{})
 	EventService().Log(containerAttachEvent, eventtypes.ContainerEventType, actor)
-	err = c.containerProxy.AttachStreams(context.Background(), vc, clStdin, clStdout, clStderr, ca)
+
+	if vc.Config.Tty {
+		ca.UseStderr = false
+	}
+
+	ac := &AttachConfig{
+		ID: id,
+		ContainerAttachConfig: ca,
+		UseTty:                vc.Config.Tty,
+		CloseStdin:            vc.Config.StdinOnce,
+	}
+
+	err = c.containerProxy.AttachStreams(context.Background(), ac, stdin, stdout, stderr)
 	if err != nil {
 		if _, ok := err.(DetachError); ok {
+			log.Infof("Detach detected, tearing down connection")
+
 			// fire detach event
-			actor := CreateContainerEventActorWithAttributes(vc, map[string]string{})
 			EventService().Log(containerDetachEvent, eventtypes.ContainerEventType, actor)
 
-			log.Infof("Detach detected, tearing down connection")
-			client = c.containerProxy.Client()
-			handle, err = c.Handle(id, name)
-			if err != nil {
-				return err
-			}
-
-			unbind, err := client.Interaction.InteractionUnbind(interaction.NewInteractionUnbindParamsWithContext(ctx).
-				WithConfig(&models.InteractionUnbindConfig{
-					Handle: handle,
-				}))
-			if err != nil {
-				return InternalServerError(err.Error())
-			}
+			// DON'T UNBIND FOR NOW, UNTIL/UNLESS REFERENCE COUNTING IS IN PLACE
+			// This avoids cutting the communication channel for other sessions connected to this
+			// container
 
-			handle, ok = unbind.Payload.Handle.(string)
-			if !ok {
-				return InternalServerError("type assertion failed")
-			}
+			// handle, err := c.Handle(id, name)
```

Comment:
It is commented out for now cause we no longer call unbind on detach, we can enable that once we have the ref counting in place. I know we can always find that snippet from git but since this change is too big I thought it is OK to leave it like that for now